File: src/com/ticker/model/Ticker.java
package com.ticker.model;

public class Ticker {
	private String symbol;
	private double price, open, high, low, change, changePer;
	private boolean direction, increase;
	

	public Ticker(String symbol, double open) {
		this.symbol = symbol;
		this.price = this.open = this.high = this.low = open;
		this.change = this.changePer = 0.0f;
	}

	public double getPrice() {
		return price;
	}

	public void setPrice(double price) {
		this.increase = price > this.price;
		this.direction = price > this.open;
		this.high = price > this.high? price: this.high;
		this.low = price < this.low? price: this.low;
		this.change = price - this.open;
		this.changePer = this.change/this.open*100;
		this.price = price;
	}

	public boolean isIncrease() {
		return increase;
	}
	
	public String getSymbol() {
		return symbol;
	}

	public double getOpen() {
		return open;
	}

	public double getHigh() {
		return high;
	}

	public double getLow() {
		return low;
	}

	public double getChange() {
		return change;
	}

	public double getChangePer() {
		return changePer;
	}

	public boolean isDirection() {
		return direction;
	}
	
	
}

File: WebContent/app/model/Ticker.js
Ext.define('StockTicker.model.Ticker', {
	extend: 'Ext.data.Model',
	fields:[
	   'symbol', 'direction',
	   {
		   name: 'price', 
		   convert: function(value) {
			   return Ext.util.Format.number(value, '0.00');
		   }
	   }, {
		   name: 'open',
		   convert: function(value) {
			   return Ext.util.Format.number(value, '0.00');
		   }
	   }, {
		   name: 'high',
		   convert: function(value) {
			   return Ext.util.Format.number(value, '0.00');
		   }
	   }, {
		   name: 'low',
		   convert: function(value) {
			   return Ext.util.Format.number(value, '0.00');
		   }
	   }, {
		   name: 'change',
		   convert: function(value) {
			   var val =  Ext.util.Format.number(value, '0.00');
			   return val <= 0? val: "+"+val;
		   }
	   }, {
		   name: 'changePer',
		   convert: function(value) {
			   var val =  Ext.util.Format.number(value, '0.00');
			   return val <= 0? val: "+"+val;
		   }
	   }
	],
	proxy: {
    	type: 'memory'
    },
    idProperty: 'symbol'
});

'open', 'high', 'low', 'direction', 'change', 'changePer'